OZON.ru.js

```diff
@@ -1,604 +1,103 @@
-		case "шестое":
-			return "6";
-		case "седьмое":
-			return "7";
-		case "восьмое":
-			return "8";
-		case "девятое":
-			return "9";
-		case "десятое":
-			return "10";
-		default:
-			return numStr;
-		}
+		if (property.name === 'Переводчик') book.creators.push(ZU.cleanAuthor(property.value, 'translator', false));
+		if (property.name === 'Редактор') book.creators.push(ZU.cleanAuthor(property.value, 'editor', false));
+		if (property.name === 'Серия') book.series = property.value;
 	}
-	return "";
-}
 
-function ParseProductDetail(pdXPathRes) {
-	var result = new Object();
-	var creatorsObject = new Object();
-	var pdDOMNode;
-	while (pdDOMNode = pdXPathRes.iterateNext()) {
-		var text = CleanText(pdDOMNode.textContent);
-		var name = text.replace(/[:,\s].*$/, "");
-		var value = text.replace(/^\S*\s+/, "");
-		//Zotero.debug(name + " -> " + value);
-		switch (name) {
-		case "Автор":
-		case "Авторы":
-			creatorsObject["authors"] = GetCreators(value, "author");
-			break;
-		case "Составитель":
-			creatorsObject["compilers"] = GetCreators(value, "editor");
-			break;
-		case "Переводчик":
-		case "Переводчики":
-			creatorsObject["translators"] = GetCreators(value, "translator");
-			break;
-		case "Редактор":
-		case "Редакторы":
-			creatorsObject["editors"] = GetCreators(value, "editor");
-			break;
-		case "Язык":
-		case "Языки":
-			result["language"] = GetLanguage(value);
-			break;
-		case "Издательство":
-			result["publisher"] = value;
-			break;
-		case "Серия":
-			result["series"] = value;
-			break;
-		case "ISBN":
-			result["ISBN"] = GetISBN(value);
-			result["date"] = GetDate(value);
-			break;
-		default:
-			//"Naked" year
-			if (name.match(/^\d\d\d\d$/) && !result["date"]) {
-				result["date"] = name;
-			}
-		}
-	}
-	creatorsObject["all"] = new Array();
-	if(creatorsObject["authors"]) {
-		creatorsObject["all"] =
-		creatorsObject["all"].concat(creatorsObject["authors"]);
-	}
-	if(creatorsObject["compilers"]) {
-		creatorsObject["all"] =
-		creatorsObject["all"].concat(creatorsObject["compilers"]);
-	}
-	if(creatorsObject["translators"]) {
-		creatorsObject["all"] =
-		creatorsObject["all"].concat(creatorsObject["translators"]);
-	}
-	if(creatorsObject["editors"]) {
-		creatorsObject["all"] =
-		creatorsObject["all"].concat(creatorsObject["editors"]);
-	}
-	if(creatorsObject["all"]) {
-		result["creators"] = creatorsObject["all"];
-	}
-	return result;
+	book.complete();
 }
 
-function ParseTechDesc(nameXPathRes, valueXPathRes) {
-	var result = Object();
-	var nameDOMNode;
-	var valueDOMNode;
-	while( (nameDOMNode = nameXPathRes.iterateNext()) &&
-	(valueDOMNode = valueXPathRes.iterateNext()) ) {
-		var nameText = CleanText(nameDOMNode.textContent);
-		var valueText = CleanText(valueDOMNode.textContent);
-		if (nameText == "Страниц") {
-			result["numPages"] = GetNumPages(valueText);
-			return result;
-		}
-	}
-}
+function getAdditionalProperty(element) {
+	var matches = /[А-я]+(?=:)/.exec(element.textContent);
```

Comment:
My mistake, did not put in the effort to check on my phone that the first letter was CYRILLIC CAPITAL A. Thanks for that @avram.

When using non-ASCII characters, particularly in ranges or where ambiguous, we should use Unicode escapes, so `[\u0410-\u044F]` (plus what @avram says above about the missing characters, though they don't matter in this case, since we're only looking at a set number of `itemprop` names)